fetcher.go

```diff
@@ -16,10 +16,16 @@ import (
 	"github.com/ipld/go-ipld-prime/traversal/selector/builder"
 )
 
-// TODO: need to support sessions
+type FetcherConfig struct {
+	blockService blockservice.BlockService
+}
 
 type Fetcher struct {
```

Comment:
Maybe this should just be `Session` now. `FetcherConfig` could become `Fetcher`. We also may want to make it private and give it an interface, but that trades safety for difficulty navigating, so maybe not.